app/actions/handlers.py

```diff
@@ -107,12 +102,19 @@ async def action_pull_observations(integration, action_config: client.PullObserv
         })
         raise e
     else:
-        logger.info(f"-- Observations pulled with success. --")
-
-        # Extract GMT offsets from transmissions
-        gmt_offsets = extract_gmt_offsets(transmissions)
-
-        logger.info(f"-- GMT offsets: {gmt_offsets} --")
+        logger.info(f"-- Observations pulled with success for integration ID: {str(integration.id)}. --")
+
+        if transmissions:
+            # Extract GMT offsets from transmissions
+            gmt_offsets = extract_gmt_offsets(transmissions)
+        else:
+            logger.warning(f"No transmissions were pulled for integration ID: {str(integration.id)}.")
+            logger.warning(f"-- Setting GMT offset to 0 for devices: {data_points_per_device.keys()} integration ID: {str(integration.id)}.")
+            gmt_offsets = {}
+            for serial_num in data_points_per_device.keys():
+                gmt_offsets.setdefault(serial_num, 0)
```

Comment:
you probably don't need this, since when you read it you are already using get and defaulting to zero: `gmt_offsets.get(serial_num, 0)`